app/vmauth/auth_config.go

```diff
@@ -398,7 +398,7 @@ func (up *URLPrefix) discoverBackendAddrsIfNeeded() {
 	}
 
 	pbus := up.bus.Load()
-	if areEqualBackendURLs(*pbus, busNew) {
+	if len(busNew) == 0 || areEqualBackendURLs(*pbus, busNew) {
```

Comment:
We shouldn't add this check, it will skip storing new backend urls.

Reply:
如果 busNew 为空数组的话，后续访问会引发 panic，导致程序退出，正常情况应该是请求失败，返回503，可以再 getBackendUrl 那里加个校验

Reply:
>如果 busNew 为空数组的话，后续访问会引发 panic，导致程序退出，

After discussion, this panic is needed in the code because:
1. in the current code, up.bus.Load() should never get nil, as we always store addresses no matter the resolve is successfully or not.
2. if up.bus.Load()==nil, code must contains bug and it should be exposed and fixed.
The change has been included in https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6401/commits/28271813a2abe396f2f7f8f10619991582599cce.